File: src/main/java/catwalks/proxy/CommonProxy.java
package catwalks.proxy;

import catwalks.item.ItemDecoration;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.server.MinecraftServer;
import net.minecraftforge.event.entity.player.EntityItemPickupEvent;
import net.minecraftforge.fml.common.eventhandler.Event.Result;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.server.FMLServerHandler;

public class CommonProxy {
	public void preInit() {}
	
	public void reloadConfigs() {}
	
	public MinecraftServer getServer() {
		return FMLServerHandler.instance().getServer();
	}
	
	@SubscribeEvent
	public void itemPickup(EntityItemPickupEvent event) {
		ItemStack stack = event.getItem().getEntityItem();
		if(event.getEntityPlayer() == null) {
			return;
		}
		InventoryPlayer inv = event.getEntityPlayer().inventory;
		boolean changed = false;

		if(stack != null && stack.getItem() instanceof ItemDecoration) {
			for (int i = 0; i < inv.getSizeInventory(); i++) {
				ItemStack slotStack = inv.getStackInSlot(i);
				if(slotStack != null && slotStack.getItem() == stack.getItem() && ( slotStack.getItemDamage() != 0 || event.getEntityPlayer().capabilities.isCreativeMode)) {
					int toTake = slotStack.getItemDamage();
					int available = stack.getMaxDamage()-stack.getItemDamage();
					if(event.getEntityPlayer().capabilities.isCreativeMode)
						toTake = available;
					int toput = Math.min(toTake, available);
					
					slotStack.setItemDamage(slotStack.getItemDamage()-toput);
					stack.setItemDamage(stack.getItemDamage()+toput);
					changed = true;
				}
			}
			if(stack.getItemDamage() == stack.getMaxDamage())
				stack.stackSize--;
			
			if(changed) {
				event.setResult(Result.ALLOW);
				event.getEntityPlayer().inventoryContainer.detectAndSendChanges();
			}
		}
		
	}
}

File: src/main/java/catwalks/proxy/CommonProxy.kt
package catwalks.proxy

import catwalks.item.ItemDecoration
import net.minecraft.entity.player.InventoryPlayer
import net.minecraft.item.ItemStack
import net.minecraft.server.MinecraftServer
import net.minecraftforge.event.entity.player.EntityItemPickupEvent
import net.minecraftforge.fml.common.eventhandler.Event.Result
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraftforge.fml.server.FMLServerHandler

open class CommonProxy {
    open fun preInit() {
    }

    open fun reloadConfigs() {
    }

    open val server: MinecraftServer
        get() = FMLServerHandler.instance().server

    @SubscribeEvent
    fun itemPickup(event: EntityItemPickupEvent) {
        val stack = event.item.entityItem
        if (event.entityPlayer == null) {
            return
        }
        val inv = event.entityPlayer.inventory
        var changed = false

        if (stack != null && stack.item is ItemDecoration) {
            for (i in 0..inv.sizeInventory - 1) {
                val slotStack = inv.getStackInSlot(i)
                if (slotStack != null && slotStack.item === stack.item && (slotStack.itemDamage != 0 || event.entityPlayer.capabilities.isCreativeMode)) {
                    var toTake = slotStack.itemDamage
                    val available = stack.maxDamage - stack.itemDamage
                    if (event.entityPlayer.capabilities.isCreativeMode)
                        toTake = available
                    val toput = Math.min(toTake, available)

                    slotStack.itemDamage = slotStack.itemDamage - toput
                    stack.itemDamage = stack.itemDamage + toput
                    changed = true
                }
            }
            if (stack.itemDamage == stack.maxDamage)
                stack.stackSize--

            if (changed) {
                event.result = Result.ALLOW
                event.entityPlayer.inventoryContainer.detectAndSendChanges()
            }
        }

    }
}
